File: oop/java/src/Lista01/Ex06/Ex06.java
package Lista01.Ex06;

import Lista01.KeyboardInput;

public class Ex06 {
    private Boolean isPrime(Integer n) {
        Integer divisor = 1;
        while(n % ++divisor != 0 && divisor < ((int) Math.sqrt(n)));
        return n % divisor != 0;
    }

    public Integer getPrimePredecessor(Integer n) {
        while (n > 1) if (isPrime(--n)) return n;
        return null;
    }

    public static void main(String[] args) {
        Ex06 predecessorPrimeCalculator = new Ex06();
        KeyboardInput keyboardInput = new KeyboardInput(System.in);

        System.out.println("Provide the base number: ");
        Integer input = keyboardInput.readPositiveInteger();

        Integer divisor = predecessorPrimeCalculator.getPrimePredecessor(input);
        System.out.println(divisor + " is the first prime number lower than the input.");
    }
}

File: oop/python/lista01/Ex06.py
from math import ceil
from typing import Union

from KeyboardInput import KeyboardInput

class Ex06:
    def isPrime(self, n) -> bool:
        divisor = 2
        while n % divisor != 0 and divisor < ceil(n**(1/2)): divisor += 1
        return n % divisor != 0

    def getPrimePredecessor(self, n) -> Union[int, None]:
        while (n > 1):
            n -= 1
            if (self.isPrime(n)): return n
        return None

if __name__ == "__main__":
    predecessorPrimeCalculator = Ex06()
    keyboardInput = KeyboardInput()

    input = keyboardInput.readPositiveInt("Provide the base number: ")

    divisor = predecessorPrimeCalculator.getPrimePredecessor(input)
    print("{0} is the first prime number lower than the input.".format(divisor))